<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>781</width>
      <height>671</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QPushButton" name="startButton">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="stopButton">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Size</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="slider_size">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="value">
            <number>80</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>SpeedX</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="slider_speedx">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>SpeedY</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="slider_speedy">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_10">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Edges</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="sliderEdges">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>5</number>
           </property>
           <property name="maximum">
            <number>40</number>
           </property>
           <property name="pageStep">
            <number>5</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="MyGLWidget" name="openGLWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>MyGLWidget</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clicked_Start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clicked_Stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderEdges</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slide_Edges()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>638</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_speedx</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slide_SpeedX()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_speedy</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slide_SpeedY()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider_size</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slide_Size()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>clicked_Start()</slot>
  <slot>clicked_Stop()</slot>
  <slot>slide_Size()</slot>
  <slot>slide_SpeedX()</slot>
  <slot>slide_SpeedY()</slot>
  <slot>slide_Edges()</slot>
 </slots>
</ui>
